### ISSUE TITLE
pseudographics broken after upgrade

### ISSUE THREAD

**Issue by nuclight** [jrmarino/synth#221]
After upgrading ports/pkgs from Jan 2023 to 12-EOL (same 12.4 underhood) synth outputs somewhat unreadable in both xterm and vt(4) console:

![image](https://github.com/jrmarino/synth/assets/24194952/b7d88480-2ce8-491b-bd2e-50418c49f2de)

**Comment by jrmarino** [comment]
This is the only complain I've seen on this (in over 10 months).  Can I assume this resolved itself?

**Comment by nuclight** [comment]
No, it didn't. I did not update system from that time due to unworking synth, though :) I can provide some debug, just don't know what.

**Comment by jrmarino** [comment]
you can try this to restore the system:

https://github.com/Ravenports/Ravenports/wiki/Synth_Install

another thing you can do is load the /usr/local/etc/synth/synth.ini and change 
```
Display_with_ncurses= true
```
to
```
Display_with_ncurses= false
```
on the profile that's active.

you won't get the ncurses interface but you should be able to build.  There's a web interface though.

**Comment by jrmarino** [comment]
to be clear, this is some kind of termcap/terminfo + ncurses issue.

**Comment by nuclight** [comment]
In fact it does build, just interface is unusable, so I use it for `synth install some_small|port` blindly, but don't feel it comfortable to be blind for large upgrades - when there is often need to press keys for shutdown.

The `false` setting works, though I did not understand on short port
```
00:00:06 => [01] Builder launched
00:00:06 => [01]          Kickoff sysutils/wait_on
00:00:06 => [02]          Shutting down
00:00:16 => [01] 00:00:10 Success sysutils/wait_on
00:00:16 => [01]          Shutting down
```
if that will have time/line updates on each build phase?

**Comment by nuclight** [comment]
> to be clear, this is some kind of termcap/terminfo + ncurses issue.

It doesn't seem to be linked with ncurses:

```
$ ldd `which synth`
/usr/local/bin/synth:
        libthr.so.3 => /lib/libthr.so.3 (0x800628000)
        libc.so.7 => /lib/libc.so.7 (0x800656000)
$ ll `which synth`
-r-xr-xr-x  1 root  wheel  1980264  8 Jan   2024 /usr/local/bin/synth*
$ file `which synth`        
/usr/local/bin/synth: ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, for FreeBSD 12.4, FreeBSD-style, stripped
$ pkg info synth | head -19
synth-2.13_1
Name           : synth
Version        : 2.13_1
Installed on   : Sun Jan 14 05:32:23 2024 MSK
Origin         : ports-mgmt/synth
Architecture   : FreeBSD:12:amd64
Prefix         : /usr/local
Categories     : ports-mgmt
Licenses       : ISCL
Maintainer     : fluffy@FreeBSD.org
WWW            : https://github.com/jrmarino/synth
Comment        : Custom package repository builder for FreeBSD and DragonFly
Options        :
        WATCHDOG       : off
Annotations    :
        FreeBSD_version: 1204000
        repo_type      : binary
        repository     : Synth
Flat size      : 1.96MiB
```

**Comment by nuclight** [comment]
`strings /usr/local/bin/synth` gives, amongst `terminal_interface-curses*adb*`, 
```
ncurses 6.4.20221231
```
for this version, and
```
ncurses 6.3.20211021
```
from `-rw-r--r--  1 root  wheel  732132  5 Dec 2022 /var/cache/pkg/synth-2.12~99b0d6aa51.pkg`

**Comment by jrmarino** [comment]
the freebsd ports version of synth should link to the base ncurses (so it won't be listed on the manifest).

The ravenports version of synth links to the ravenports-provided terminfo.


Again - I recommended pointing a browser at the web-based reported if you desire details during the build.

also, if you can't fix the base terminfo problem, I do recommend you install the ravenports version of synth (which is currently newer than the freebsd ports version).  I believe there's something wrong with your base but you can work around it with either using web-based reports or the ravenports-built synth.

**Comment by nuclight** [comment]
I see that synth 3 is released and there were curses discussion/patches in PRs referenced from freshports commit logs (as far as I understoo, that was ports ncurses statically linked, not base in my case), so chances are high that problem no longer exists. I probably just need to install fresh synth first somehow, before upgrading all other ports with it, when I'll receive new videocard in a week and start base/pkg builds & upgrades.

**Comment by nuclight** [comment]
Not reproduced on lowest supported FreeBSD 13.4 with Synth 3.03.
